tests/lib/azure/subnet.go

```diff
@@ -0,0 +1,44 @@
+package azure
+
+import (
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/NeowayLabs/klb/tests/lib/azure/fixture"
+)
+
+type Subnet struct {
+	client network.SubnetsClient
+	f      fixture.F
+}
+
+func NewSubnet(f fixture.F) *Subnet {
+	as := &Subnet{
+		client: network.NewSubnetsClient(f.Session.SubscriptionID),
+		f:      f,
+	}
+	as.client.Authorizer = f.Session.Token
+	return as
+}
+
+// AssertExists checks if subnet exists in the resource group.
+// Fail tests otherwise.
+func (s *Subnet) AssertExists(t *testing.T, vnetName, subnetName, address, nsg string) {
+	s.f.Retrier.Run(newID("Subnet", "AssertExists", subnetName), func() error {
+		subnet, err := s.client.Get(s.f.ResGroupName, vnetName, subnetName, "")
+		if err != nil {
+			return err
+		}
+		addressSubnet := *subnet.SubnetPropertiesFormat.AddressPrefix
```

Comment:
Now I remembered that all fields on the struct are pointers...If I'm not wrong...on this case validating will be a little annoying. This is definitely not a blocker...just to open the discussion. The tradeoff is to annoy the code writer now or the debugger later.